src/garage/_primitives.py

```diff
@@ -0,0 +1,152 @@
+"""Policy interface."""
+import abc
+
+
+class Primitive(abc.ABC):
+    """Primitive interface.
+
+    Args:
+        name (str): Name for logs, variable scopes, etc.
+        spec (garage.InOutSpec): Input/output specification this primitive will
+            satisfy.
+
+    """
+
+    def __init__(self, name, spec):
+        self._name = name
+        self._spec = spec
+
+    @abc.abstractmethod
+    def get_parameters(self):
+        """Get parameters of this primitive.
+
+        Returns:
+            dict[str, np.ndarray]: A dictionary mapping from names to values of
+                each parameter.
+
+        """
+
+    @abc.abstractmethod
+    def set_parameters(self, parameters):
+        """Set the parameters of this primtive.
+
+        Args:
+            parameters (dict[str, np.ndarray]): A dictionary mapping from names
+                to values of each parameter.
+
+        """
+
+    @property
+    def spec(self):
+        """InOutSpec: The input/output spec for this primitive."""
+        return self._spec
+
+    @property
+    def name(self):
+        """str: Name of this primitive."""
+        return self._name
+
+    @abc.abstractmethod
+    def clone(self, name):
```

Comment:
Is there a reason (besides Tensorflow 1.0 being weird) that we should include this in the API? Why can't we just use the `copy.deepcopy` API?

Reply:
`name` is needed because two different variables cannot exist in the same variable scope (i.e. same name). Is this not happening anymore in TF2?